test/java/controllers/KairosDbProxyControllerTest.java

```diff
@@ -0,0 +1,197 @@
+        assertEquals("All queried metrics must have at least one aggregator",
+                Helpers.contentAsString(result));
+
+        // ***
+        // Test success case where both metrics have aggregators
+        // ***
+        metric2Builder.setAggregators(ImmutableList.of(new Aggregator.Builder().setName("sum").build()));
+        builder.setMetrics(ImmutableList.of(metric1Builder.build(), metric2Builder.build()));
+
+        request = Helpers.fakeRequest()
+                .method(Helpers.POST)
+                .uri("/api/v1/datapoints/query")
+                .header("Content-Type", "application/json")
+                .bodyJson(OBJECT_MAPPER.<JsonNode>valueToTree(builder.build()));
+
+        when(_mockKairosDbClient.queryMetrics(any())).thenReturn(
+                CompletableFuture.completedFuture(new MetricsQueryResponse.Builder()
+                        .setQueries(ImmutableList.of()).build())
+        );
+        result = Helpers.invokeWithContext(request, Helpers.contextComponents(), () -> {
+            final CompletionStage<Result> completionStage = _controller.queryMetrics();
+            return completionStage.toCompletableFuture().get(10, TimeUnit.SECONDS);
+        });
+
+        assertEquals(Http.Status.OK, result.status());
+        assertEquals("{\"queries\":[]}", Helpers.contentAsString(result));
+    }
+
+    @Test
+    public void testQueryRequiresAggregatorOff() {
+        when(_mockConfig.getBoolean(eq("kairosdb.proxy.requireAggregators"))).thenReturn(false);
+
+        final KairosDbProxyController controller = new KairosDbProxyController(
+                _mockConfig,
+                _mockWSClient,
+                _mockKairosDbClient,
+                OBJECT_MAPPER,
+                _mockMetricsFactory,
+                _mockMetricsqueryConfig
+        );
+
+        final Metric.Builder metric1Builder = new Metric.Builder()
+                .setName("metric1")
+                .setAggregators(ImmutableList.of(new Aggregator.Builder().setName("count").build()));
+        final Metric.Builder metric2Builder = new Metric.Builder()
+                .setName("metric2");
+        final MetricsQuery.Builder builder = new MetricsQuery.Builder()
+                .setStartTime(Instant.now())
+                .setMetrics(ImmutableList.of(metric1Builder.build(), metric2Builder.build()));
+
+        final Http.RequestBuilder request = Helpers.fakeRequest()
+                .method(Helpers.POST)
+                .uri("/api/v1/datapoints/query")
+                .header("Content-Type", "application/json")
+                .bodyJson(OBJECT_MAPPER.<JsonNode>valueToTree(builder.build()));
+
+        when(_mockKairosDbClient.queryMetrics(any())).thenReturn(
+                CompletableFuture.completedFuture(new MetricsQueryResponse.Builder()
+                        .setQueries(ImmutableList.of()).build())
+        );
+
+        // ***
+        // Test failure case where one metric doesn't have an aggregator
```

Comment:
nit. I wouldn't necessarily call this a failure case any more since it'd work fine with this `requiresAggregators` configuration turned off but I follow